joomla/joomla-cms
# Multilanguage: correcting saving menu item with same alias when language is changed to ALL

### Summary of Changes
Making a new menu item set to ALL or changing an existing menu item to ALL act differently when the same alias already exists (at Root).

### Testing Instructions
On a multingual site, one can use the same alias for menu items in the root of their specific language as long as these menu items are tagged to the specific different languages, NOT including All.
i.e one can have mysite.com/en/isaacnewton and mysite,com/fr/isaacnewton.

To test, create a simple multilingual site
Create in fr-FR a menu item with the alias isaacnewton tagged to fr-FR at the root of the menu.
Create in an en-GB menu a menu item tagged to ALL with the same alias isaacnewton at the root of the menu.
You can't save the menu item. All is good.

Now change the language to en-GB. It saves OK and that is fine.

Now, edit again that menu item and change its language to ALL: **it is possible although it should NOT be!**

Patch and test again.

In that specific case (changing to ALL), one will get the message:
`Save failed with the following error: The alias search is already being used by the search menu item in another menu (remember it may be a trashed item).`

We can't use the normal message as it could be confusing by containing the `menutype` while we change a menu item to ALL in the same menu type.
### Documentation Changes Required
None. This is a bug
